### DISCUSSION TITLE
Unlimited animation prevents application to close on Linux

### DISCUSSION THREAD

**Discussion by llfab** [AvaloniaUI/Avalonia#6965] [Q&A]
I use an INFINITE animation within my main window. This is not a problem on Windows. When I close the window, the application terminates correctly. On Linux, the window close seems to be blocked when the animation is still running. I tried this using an somewhat long animation: while it was running the window would not close, if the animation has ended, the window would close correctly.

How do I terminate all animation on window close?

**Comment by llfab** [comment]
In detail: it is a key frame animation!

**Reply by kekekeks** [reply]
Do you have a simple repro project?

**Comment by llfab** [comment]
Yes, I created a very minimum App. See attachment. I noticed: if close button is hit on Linux, the animation stops but the window does not close.
[AvaloniaApp.zip](https://github.com/AvaloniaUI/Avalonia/files/7554564/AvaloniaApp.zip)

**Comment by llfab** [comment]
Note, the App has a looong but not an infinite animation which is good enough to reproduce

**Comment by llfab** [comment]
I do see the very same behavior with the OpenGLPage from the Avalonia Controls sample implementation. Once the "Disco" animation is on, the app does not close on Linux. If I bring the "Disco" slider back to the very left, the application closes correctly.

**Reply by kekekeks** [reply]
That's interesting since that "disco" animation doesn't use animations at all, it just calls `Dispatcher.UIThread.Invoke(()=>InvalidateVisual())`
